source/solvers/mf_navier_stokes.cc

```diff
@@ -44,6 +45,209 @@
 
 #include <deal.II/numerics/vector_tools.h>
 
+template <typename VectorType>
+class PreconditionBase
```

Comment:
Is there a reason why you want to declare the class just in the .cc and not in the .h upstream? 
Also, can you add @brief and documentation for the base class following doxygen syntax.

Reply:
> Is there a reason why you want to declare the class just in the .cc and not in the .h upstream?

I would like to keep this local here as long as it is not needed anywhere else.

> Also, can you add @brief and documentation for the base class following doxygen syntax.

Done!